### PR TITLE
Scenario runner

### PR DESCRIPTION
**Pull request by AleksandarFilipov**
Allows list to be provided to be executed.

### CODE REVIEW

**Comment by AleksandarFilipov** [comment]
This merge is kind of big, so please bare with me.

File: apps/ae_socket_connector/lib/channel_runner.ex [hunk 1]
```diff
@@ -7,7 +7,8 @@ defmodule ChannelRunner do
   @pause 2000
 
   def start_channel_helper(),
-    do: start_channel_helper(:single)
+    # do: start_channel_helper(:single)
+    do: ClientRunner.start_helper()
```
Comment (tolbrino) [inline apps/ae_socket_connector/lib/channel_runner.ex:11]: `start_channel_helper/0` and `start_channel_helper/1` seem to have very different semantics now. Should they have different function names?

File: apps/ae_socket_connector/lib/client_runner.ex [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  use GenServer
+  require Logger
+
+  defstruct pid_session_holder: nil,
+            color: nil,
+            job_list: nil
+
+  def start_link(
+        {_pub_key, _priv_key, %SocketConnector.WsConnection{}, _ae_url, _network_id, _role, _jobs,
+         _color} = params
```
Comment (tolbrino) [inline apps/ae_socket_connector/lib/client_runner.ex:11]: `params` should be a map, which helps make the API more readable. Also why are you checking only for the struct here?

File: apps/ae_socket_connector/lib/client_runner.ex [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+      ) do
+    GenServer.start_link(__MODULE__, params)
+  end
+
+  # Server
+  def init(
+        {pub_key, priv_key, %SocketConnector.WsConnection{} = state_channel_configuration, ae_url,
+         network_id, role, jobs, color}
+      ) do
```
Comment (tolbrino) [inline apps/ae_socket_connector/lib/client_runner.ex:20]: Same as above.

File: apps/ae_socket_connector/lib/client_runner.ex [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    current_pid = self()
+
+    {:ok, pid_session_holder} =
```
Comment (tolbrino) [inline apps/ae_socket_connector/lib/client_runner.ex:23]: Way too much nested cases. Please refactor.

File: apps/ae_socket_connector/lib/client_runner.ex [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+    {:ok,
+     %__MODULE__{
+       pid_session_holder: pid_session_holder,
+       job_list: jobs,
+       color: [ansi_color: color]
+     }}
+  end
+
+  def handle_cast({:process_job_lists}, state) do
+    remaining_jobs =
```
Comment (tolbrino) [inline apps/ae_socket_connector/lib/client_runner.ex:80]: Please reduce nested cases.

File: apps/ae_socket_connector/lib/socket_connector.ex [hunk 6]
```diff
@@ -905,35 +946,36 @@ defmodule SocketConnector do
...
-           Validator.get_state_round(to_sign) => %Update{
-             updates: updates,
-             tx: to_sign,
-             contract_call: state.contract_call_in_flight
-           }
-         },
-         contract_call_in_flight: nil
-     }}
-  end
+  # def process_message(
```
Comment (tolbrino) [inline apps/ae_socket_connector/lib/socket_connector.ex:949]: Please remove unused code. This is what Git is for.

**Review by tolbrino — APPROVED** [review]
This change deserves more documentation. In general I think you should focus on adding module and function documentation as this will help reviews and using the framework. I won't block this PR, but take my comments as suggestions for improvements for the next PR.

**Pull request merged**